scripts/aviti_run_parameter_parser.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python
+
+import glob
+import json
+import os
+import sys
+from argparse import ArgumentParser
+from datetime import datetime
+
+from genologics.config import BASEURI, PASSWORD, USERNAME
+from genologics.entities import Process
+from genologics.lims import Lims
+
+DESC = """EPP for parsing run paramters for AVITI runs
+Author: Chuan Wang, Science for Life Laboratory, Stockholm, Sweden
+"""
+
+
+def fetch_fc(process):
+    fc_id = ""
+    if "Load to Flowcell (AVITI)" in process.parent_processes()[0].type.name:
+        fc_id = process.parent_processes()[0].output_containers()[0].name
+    else:
+        sys.stderr.write("No associated parent step can be found.")
+        sys.exit(2)
+    return fc_id
+
+
+def fetch_rundir(fc_id):
+    run_dir = ""
+    metadata_dir = "ngi-nas-ns"
+    data_dir = "AVITI_data"
+    run_dir_path = os.path.join(os.sep, "srv", metadata_dir, data_dir, f"*{fc_id}")
+    if len(glob.glob(run_dir_path)) == 1:
+        run_dir = glob.glob(run_dir_path)[0]
+    elif len(glob.glob(run_dir_path)) == 0:
+        sys.stderr.write(f"No run dir can be found for FC {fc_id}")
+        sys.exit(2)
+    else:
+        sys.stderr.write(f"Multiple run dirs found for FC {fc_id}")
+        sys.exit(2)
+    return run_dir
+
+
+def attach_json_files(process, run_dir):
+    for outart in process.all_outputs():
+        if outart.type == "ResultFile" and outart.name == "Run Parameters":
+            try:
+                lims.upload_new_file(outart, f"{run_dir}/RunParameters.json")
+            except OSError:
+                try:
+                    lims.upload_new_file(outart, f"{run_dir}/RunParameters.json")
```

Comment:
Are we trying the same thing twice just in case the connection is bumpy?

Reply:
Thanks for pointing this up! The redundant lines have been deleted.